Tests/SwiftParserTest/translated/AsyncTests.swift

```diff
@@ -27,94 +27,76 @@ final class AsyncTests: XCTestCase {
+      ],
+      fixedSource: "func asyncGlobal3() async throws { }"
     )
   }
 
   func testAsync3() {
     AssertParse(
-      """
-      func asyncGlobal3(fn: () throws -> Int) rethrows1️⃣ async { }
-      """,
+      "func asyncGlobal3(fn: () throws -> Int) rethrows 1️⃣async { }",
       diagnostics: [
-        DiagnosticSpec(message: "consecutive statements on a line must be separated by ';'")
-        // TODO: Old parser expected error on line 1: 'async' must precede 'rethrows', Fix-It replacements: 50 - 56 = '', 41 - 41 = 'async '
-      ]
+        DiagnosticSpec(message: "'async' must precede 'rethrows'", fixIts: ["move 'async' in front of 'rethrows'"])
+      ],
+      fixedSource: "func asyncGlobal3(fn: () throws -> Int) async rethrows { }"
     )
   }
 
   func testAsync4() {
     AssertParse(
-      """
-      func asyncGlobal4() -> Int1️⃣ async { }
-      """,
+      "func asyncGlobal4() -> Int 1️⃣async { }",
       diagnostics: [
-        DiagnosticSpec(message: "consecutive statements on a line must be separated by ';'")
-        // TODO: Old parser expected error on line 1: 'async' may only occur before '->', Fix-It replacements: 28 - 34 = '', 21 - 21 = 'async '
-      ]
+        DiagnosticSpec(message: "'async' may only occur before '->'", fixIts: ["move 'async' in front of '->'"])
+      ],
+      fixedSource: "func asyncGlobal4() async -> Int { }"
     )
   }
 
   func testAsync5() {
     AssertParse(
-      """
-      func asyncGlobal5() -> Int1️⃣ async throws 2️⃣{ }
-      """,
+      "func asyncGlobal5() -> Int 1️⃣async throws { }",
       diagnostics: [
-        // TODO: Old parser expected error on line 1: 'async' may only occur before '->', Fix-It replacements: 28 - 34 = '', 21 - 21 = 'async '
-        // TODO: Old parser expected error on line 1: 'throws' may only occur before '->', Fix-It replacements: 34 - 41 = '', 21 - 21 = 'throws '
-        DiagnosticSpec(locationMarker: "1️⃣", message: "consecutive statements on a line must be separated by ';'"),
-        DiagnosticSpec(locationMarker: "2️⃣", message: "expected '->'"),
-      ]
+        DiagnosticSpec(message: "'async throws' may only occur before '->'")
+      ],
+      fixedSource: "func asyncGlobal5() async throws -> Int { }"
     )
   }
 
   func testAsync6() {
     AssertParse(
-      """
-      func asyncGlobal6() -> Int 1️⃣throws async { }
-      """,
+      "func asyncGlobal6() -> Int 1️⃣throws async { }",
       diagnostics: [
-        // TODO: Old parser expected error on line 1: 'throws' may only occur before '->', Fix-It replacements: 28 - 35 = '', 21 - 21 = 'throws '
-        // TODO: Old parser expected error on line 1: 'async' may only occur before '->', Fix-It replacements: 35 - 41 = '', 21 - 21 = 'async '
-        DiagnosticSpec(message: "extraneous code 'throws async { }' at top level")
-      ]
+        DiagnosticSpec(message: "'throws async' may only occur before '->'")
```

Comment:
Little bit weird, but probably good enough.

Reply:
I actually quite like what the diagnostic generator spits out here.

Reply:
It's cool that it merges them, what's the fix-it message? Is it the same "Move 'throws async' before '->'", is it "Move 'async throws' before '->'", or something else?

Reply:
Yes, it produces `"move 'throws async' in front of '->'"`